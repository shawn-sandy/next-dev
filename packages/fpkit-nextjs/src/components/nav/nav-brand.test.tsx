import React from "react";

import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import { axe } from "vitest-axe";

// ðŸ‘‰ import ComponentName from "./component-name";
import NavBrand from "./nav-brand";

const brand = {
  logo: "FPKit NextJS",
  url: "/",
};

import Nav from "./nav";
describe("<NavBrand/>", () => {
  test("renders without crashing", async () => {
    render(<NavBrand url={brand.url} logo={brand.logo} />);
    const brandHeading = screen.getByRole("heading");
    expect(brandHeading).toBeDefined();
    // @ts-ignore
    expect(await axe(brandHeading)).toHaveNoViolations();
    // expect(brandHeading).
    screen.logTestingPlaygroundURL();
    expect(brandHeading).toMatchSnapshot();
  });
});
