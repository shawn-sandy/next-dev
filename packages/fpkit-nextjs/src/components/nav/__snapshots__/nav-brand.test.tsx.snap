// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NavBrand/> > renders without crashing 1`] = `
<h1
  style="--fs: 1.2rem;"
>
  <a
    href="/"
    style="text-transform: capitalize;"
  >
    FPKit NextJS
  </a>
</h1>
`;
